lib/ansible/modules/windows/win_domain_group_membership.ps1

```diff
@@ -0,0 +1,141 @@
+#!powershell
+
+# Copyright: (c) 2019, Marius Rieder <marius.rieder@scs.ch>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+try {
+    Import-Module ActiveDirectory
+}
+catch {
+    Fail-Json $result "win_domain_group_membership requires the ActiveDirectory PS module to be installed"
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$diff_mode = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+# Module control parameters
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent","pure"
+$domain_username = Get-AnsibleParam -obj $params -name "domain_username" -type "str"
+$domain_password = Get-AnsibleParam -obj $params -name "domain_password" -type "str" -failifempty ($domain_username -ne $null)
+$domain_server = Get-AnsibleParam -obj $params -name "domain_server" -type "str"
+
+# Group Membership parameters
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$members = Get-AnsibleParam -obj $params -name "members" -type "list" -failifempty $true
+
+$extra_args = @{}
+if ($domain_username -ne $null) {
+    $domain_password = ConvertTo-SecureString $domain_password -AsPlainText -Force
+    $credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $domain_username, $domain_password
+    $extra_args.Credential = $credential
+}
+if ($domain_server -ne $null) {
+    $extra_args.Server = $domain_server
+}
+
+$result = @{
+    changed = $false
+}
+if ($state -in @("present", "pure")) {
+    $result.added = @()
+}
+if ($state -in @("absent", "pure")) {
+    $result.removed = @()
+}
+if ($diff_mode) {
+    $result.diff = @{}
+}
+
+$members_before = Get-AdGroupMember -Identity $name @extra_args
+$pure_members = @()
+
+foreach ($member in $members) {
+    # Use try/catch as Get-ADUser does not support `-ErrorAction SilentlyContinue`
+    # We do not want to fail but query for group if no user is found
+    try {
+        $group_member = Get-ADUser -Identity $member @extra_args
```

Comment:
It might be better to use `Get-ADObject` instead of the specific `Get-ADUser` or `Get-ADObject`. IIRC you can set more than just groups/users as a member of a domain group but I could be wrong here.

Reply:
According to `Add-ADGroupMember` it's `users`, `groups`, `service accounts`, or `computer`. However `Get-ADObject` returns even more Object types and its `-Identity` parameter does only take the `DN ` or `GUID ` of an object so the query has a quiet long filter, however its now down to only one query without try/catch